server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/csv/IRVHeadersParser.java

```diff
@@ -0,0 +1,128 @@
+/*
+Democracy Developers IRV extensions to colorado-rla.
+
+@copyright 2024 Colorado Department of State
+
+These IRV extensions are designed to connect to a running instance of the raire 
+service (https://github.com/DemocracyDevelopers/raire-service), in order to 
+generate assertions that can be audited using colorado-rla.
+
+The colorado-rla IRV extensions are free software: you can redistribute it and/or modify it under the terms
+of the GNU Affero General Public License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+The colorado-rla IRV extensions are distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.csv;
+
+import au.org.democracydevelopers.corla.model.vote.IRVParsingException;
+import au.org.democracydevelopers.corla.model.vote.IRVPreference;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import us.freeandfair.corla.model.Choice;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Utils for parsing IRV headers as part of DominionCVRExportParser.
+ * Main methods:
+ * - validateIRVPreferenceHeaders validates a whole collection as part of a CSV header, to check
+ *   that they are consistent with the expected pattern.
+ * - generateAllIRVPreferences produces the expected pattern of preferences, given the list of
+ *   choices and the number of ranks allowed.
+ */
+public class IRVHeadersParser {
+  /**
+   * Class-wide logger
+   */
+  private static final Logger LOGGER = LogManager.getLogger(IRVHeadersParser.class);
+
+  /**
+   * Checks whether a CSVRecord contains a sequence of choices as expected for IRV - it should be
+   * a sequence of name(rank) choices, starting with preference 1 and going up to maxRank,
+   * and for each rank, it should contain all the same choices in the same order. For example,
+   * Alice(1), Bob(1), Chuan(1), Alice(2), Bob(2), Chuan(2), Alice(3), Bob(3), Chuan(3) is valid,
+   * but any rearrangement of those items, or anything that omits some rank or candidate included
+   * elsewhere, is invalid.
+   * @param theLine the CSV line (basically a list of Strings)
+   * @param startIndex the first index for this contest's choices.
+   * @param numChoices the number of choices (candidates).
+   * @param maxRank The number of ranks allowed (which matches the number expected in the csv).
```

Comment:
Should the paragraph starting at "This works ..." be moved above the param statements? As otherwise I imagine it might be associated with the @throws statement?